src/tools/opt-dist/src/environment.rs

```diff
@@ -25,6 +25,27 @@ pub struct Environment {
     prebuilt_rustc_perf: Option<Utf8PathBuf>,
     use_bolt: bool,
     shared_llvm: bool,
+    /// Additional configuration that bootstrap needs to know only when running tests.
+    #[builder(default)]
+    test_config: TestConfig,
+}
+
+/// Builds have optional components, and their presence/absence can enable/disable a subset of
+/// tests. When testing the optimized artifacts, bootstrap needs to know about these enabled
+/// components to run the expected subset. This structure holds the known components where this
+/// matters: currently only whether the build to test is using debug assertions.
+///
+/// FIXME: ultimately, this is a temporary band-aid, and opt-dist should be more transparent to the
+/// CI config and bootstrap optional components: bootstrap has default values, combinations of flags
+/// that cascade into others, etc logic that we'd have to duplicate here otherwise. It's more
+/// sensible for opt-dist to never know about the config apart from the minimal set of paths
+/// required to configure stage0 tests.
+#[derive(Builder, Default, Clone, Debug)]
+pub struct TestConfig {
+    /// Whether the build under test is explicitly using `--enable-debug-assertions`.
+    /// Note that this flag can be implied from others, like `rust.debug`, and we do not handle any
+    /// of these subtleties and defaults here, as per the FIXME above.
+    pub enable_debug_assertions: bool,
```

Comment:
Nit: maybe call this something like `has_rustc_debug_assertions` to distinguish it from the orthogonal LLVM having debug assertions or not.

Reply:
AFAICT there are no "llvm debug assertions", right? (only "llvm assertions") -- but I've renamed this to `rust_debug_assertions`  because it's the `rust.debug_assertions` config key. But there's also some value in having a name that matches the `./x` flag that we actually use.